fn_spamhaus_query/fn_spamhaus_query/components/fn_spamhaus_query_submit_artifact.py

```diff
@@ -0,0 +1,103 @@
+
+    @function("fn_spamhaus_query_submit_artifact")
+    def _fn_spamhaus_query_submit_artifact_function(self, event, *args, **kwargs):
+        """Function: Function to check IP Address & Domain Names against Spamhaus Database to see whether IP Address or Domain Names appears in Spamhaus block list records or not."""
+        try:
+            # Get the function parameters:
+            spamhaus_query_string = kwargs.get("spamhaus_query_string")  # text
+            spamhause_search_resource = kwargs.get("spamhause_search_resource")  # text
+
+            # Get the app.config parameters:
+            spamhaus_wqs_url = self.options.get("spamhaus_wqs_url")
+            spamhaus_dqs_key = self.options.get("spamhaus_dqs_key")
+
+            log = logging.getLogger(__name__)
+            log.info("spamhaus_query_string: %s", spamhaus_query_string)
+            log.info("spamhause_search_resource: %s", spamhause_search_resource)
+            log.info("spamhaus_wqs_url: %s", spamhaus_wqs_url)
+            log.info("spamhaus_dqs_key: %s", spamhaus_dqs_key)
+
+            yield StatusMessage(
+                "Checking Artifact: {} against Spamhaus block list resource {} database.".format(spamhaus_query_string,
+                                                                                                 spamhause_search_resource))
+
+            # Checking API Key
+            if not spamhaus_dqs_key:
+                raise ApikeyError("API key must be defined in App.config file Spamhaus section.")
+
+            # Initialising the Result payload object
+            result_object = ResultPayload("fn_spamhaus_query", **kwargs)
+
+            # Initialising Request Common for REST Api Call
+            request_common_obj = RequestsCommon(opts=self.options)
+
+            # Get proxy Configuration
+            proxies = request_common_obj.get_proxies()
+
+            # Construct call header with api key
+            header_data = {'Authorization': 'Bearer {}'.format(spamhaus_dqs_key)}
+
+            # Make Get Call to Spamhause website
+            response_object = request_common_obj.execute_call(verb='GET',
+                                                              url=spamhaus_wqs_url.format(spamhause_search_resource,
+                                                                                          spamhaus_query_string),
+                                                              headers=header_data, proxies=proxies,
+                                                              callback=spamhause_call_error)
+            # Get Received data in JSON format.
+            response_json = response_object.json()
+            if not response_json:
+                raise SpamhauseRequestCallError("No Response Returned from Api call")
+
+            if response_object.status_code == 404:
+                response_json['is_in_blocklist'] = False  # a bool flag to for block list status
```

Comment:
```suggestion
                response_json['is_in_blocklist'] = False  # a bool flag to for block list status
                results = result_object.done(success=False, content=response_json)
```
Should set the result payload as done here so that it accurately reports that the call failed

Youll also need to move line 99 inside the `elif` so if the response code is 404 we call `result_object.done` with success false and if 200 we set it as success = True.

Reply:
response code 404 means the given artifact is not present in the spamhaus block list. its not an error. so have done this way. please let me know your view on this.

Reply:
Thats a good point. 
My question then becomes this :
Should the payload be marked as `success:True` if the function did not find anything?ATM if a 404 is given then the function shows `success:True` which could give the impression a result was found

Reply:
what i was thinking is, the result can be either given atrifact found in spamhaus blocklist, or not found in blocklist both are valid results returned from spamhaus.
so in this context if 404 is also valid result and result payload can be `success:True`. so that api call to spamhaus is successful and results is given artifact is not a spam. 
please let me know your view on this.

Reply:
* Here I agree with Neetin because we had a discussion about it
* We want the Function to pass, even if no results were found
 * e.g. This IP address is not from a blocklist or this IP address is in a blocklist - are both correct
* In the requirements in Trello: https://trello.com/c/WrR4r0rP/29-neetin-fnspamhausquery-v100 we have stated that one of the Function Outputs should be `is_in_blocklist`
* Here the SpamHaus APIs gives a 404 if the request is correct, but the IP is not in a block list

Reply:
This makes sense. Thank you both for clarifying 👍